modules/veteran/app/models/veteran/user.rb

```diff
@@ -44,7 +44,7 @@ def bgs_service
       external_key = "#{@user.first_name} #{@user.last_name}"
       @bgs_service ||= BGS::Services.new(
         external_uid: @user.mpi_icn,
-        external_key: external_key
+        external_key: external_key.presence || @user.mpi_icn
```

Comment:
This was required due to not having a first name or last name when retrieving the Veteran from MPI using only their ICN.

Reply:
It should be fine as BGS only uses these values for logging purposes, as I understand.